tests/temp/tempFileTestInternalApi.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+require_once(dirname(__FILE__) . "/../../interface/globals.php");
+?>
+<html>
+<head>
+    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.1.1/jquery.min.js"></script>
+
+    <script language="JavaScript">
+        function testApi() {
+            $.ajax({
+                type: 'GET',
+                url: '../../apis/api/facility',
+                dataType: 'json',
+                headers: {
+                    'apicsrftoken': <?php echo js_escape($_SESSION['api_csrf_token']); ?>
+                },
+                success: function(thedata){
+                    var thedataJSON = JSON.stringify(thedata);
+                    $("#api").html(thedataJSON);
+                },
+                error:function(){
+                }
+            });
+        }
+        $(document).ready(function(){
+            testApi();
+        });
+    </script>
+
+
+</head>
+
+<?php
+// Looking into how to use the api locally via php calls (3 methods are shown below)
+
+
+// CALL via ajax Rest
```

Comment:
First use case:
Calling api from ajax using same session (ie. standard use openemr) with incorporation of csrf protection. Matrix needs this mechanism for their react work.
@sharonco-matrix , check out above, where you will just need to include the apicsrftoken entry on the header in your fetch commands; let me know if this mechanism may cause you any issues.

Reply:
Note there are no examples of this use in the main codebase yet.